pkg/rtc/mediatrack.go

```diff
@@ -65,15 +60,9 @@ type MediaTrack struct {
 
 	done chan struct{}
 
-	// quality level enable/disable
-	maxQualityLock               sync.RWMutex
-	maxSubscriberQuality         map[livekit.ParticipantID]livekit.VideoQuality
-	maxSubscribedQuality         livekit.VideoQuality
-	allSubscribersMuted          bool
-	onSubscribedMaxQualityChange func(trackID livekit.TrackID, subscribedQualities []*livekit.SubscribedQuality) error
-	maxQualityTimer              *time.Timer
-
 	onClose []func()
+
+	*MediaTrackSubscriptions
```

Comment:
I haven't seen composited types using a pointer to a struct before. Interesting that this can satisfy the interface methods.